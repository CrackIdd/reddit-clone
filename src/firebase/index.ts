import { initializeApp } from "firebase/app";
import {
  getAuth,
  onAuthStateChanged,
  GoogleAuthProvider,
  signInWithPopup,
  signOut,
} from "firebase/auth";
import {
  getFirestore,
  collection,
  addDoc,
  query,
  orderBy,
  limit,
  onSnapshot,
  setDoc,
  updateDoc,
  doc,
  serverTimestamp,
} from "firebase/firestore";
import {
  getStorage,
  ref,
  uploadBytesResumable,
  getDownloadURL,
} from "firebase/storage";
import { getMessaging, getToken, onMessage } from "firebase/messaging";
import { getPerformance } from "firebase/performance";
import app from "./firebase-config";
import { authStateObserver } from "../redux/store";

const auth = getAuth(app);

export async function signIn() {
  const provider = new GoogleAuthProvider();
  await signInWithPopup(auth, provider);
}

export async function signOutUser() {
  await signOut(auth);
}

export function initFirebaseAuth() {
  onAuthStateChanged(auth, authStateObserver);
}

export function getUserName() {
  return auth.currentUser?.displayName;
}

export function isUserSigned() {
  return !!auth.currentUser;
}

initFirebaseAuth();
