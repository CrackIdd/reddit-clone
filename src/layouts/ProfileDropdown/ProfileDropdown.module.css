.profile-dropdown__button {
  outline: 2px solid transparent;
  padding-top: 0.4rem;
  display: flex;
  align-items: center;
  column-gap: 8px;
  width: 100%;
  background-color: inherit;
  border-radius: 9999px;
  color: white;
  border: none;
}
.profile-dropdown__button:hover {
  background-color: #242424;
  cursor: pointer;
}

.profile-dropdown__icon {
  width: 24px;
  height: 24px;
  filter: brightness(0) invert(1);
}
