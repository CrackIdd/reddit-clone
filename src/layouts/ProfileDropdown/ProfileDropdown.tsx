import CSSModules from "react-css-modules";
import Dropdown from "../Dropdown/Dropdown";
import styles from "./ProfileDropdown.module.css";
import logOutIcon from "../../assets/log-out-icon.svg";

interface Props {
  dropdown: string
}

const ProfileDropdown: React.FC<Props> = ({dropdown}) => {
  return (
    <Dropdown dropdown={dropdown}>
      <div styleName="profile-dropdown">
        <button styleName="profile-dropdown__button">
          <img src={logOutIcon} alt="log out icon" />
          Log Out
        </button>
      </div>
    </Dropdown>
  );
};

export default CSSModules(ProfileDropdown, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
