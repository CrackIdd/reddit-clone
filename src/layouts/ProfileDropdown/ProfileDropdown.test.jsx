import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import ProfileDropdown from "./ProfileDropdown";
import StoreProvider from "../../redux/provider";
import * as firebase from "../../firebase";

describe("ProfileDropdown", () => {
  test("snapshot", () => {
    const { asFragment } = render(<ProfileDropdown />);

    expect(asFragment()).toMatchSnapshot();
  });

});
