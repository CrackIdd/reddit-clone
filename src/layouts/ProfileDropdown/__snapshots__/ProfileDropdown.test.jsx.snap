// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDropdown snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown profile-width"
  >
    <div
      class="dropdown__content"
    >
      <div
        class="profile-dropdown"
      >
        <button
          class="profile-dropdown__button"
        >
          <img
            alt="log out icon"
            class="profile-dropdown__icon"
            src="log-out-icon.svg"
          />
          Log Out
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
